src/backend/distributed/commands/function.c

```diff
@@ -1454,7 +1454,16 @@ DefineAggregateStmtObjectAddress(Node *node, bool missing_ok)
 	}
 	else
 	{
-		objectWithArgs->objargs = list_make1(makeTypeName("anyelement"));
+		DefElem *defItem = NULL;
+		foreach_ptr(defItem, stmt->definition)
+		{
+			/* if no explicit args are given, pg includes basetype in the signature */
+			if (strcmp(defItem->defname, "basetype") == 0 && IsA(defItem->arg, TypeName))
```

Comment:
what does it mean when basetype is not a TypeName?

Reply:
```sql
CREATE AGGREGATE newavg (
   sfunc = int4_avg_accum, basetype = int4, stype = _int8,
   finalfunc = int8_avg,
   initcond1 = '{0,0}'
);
```
For cases like the one above, PG creates the aggregate using the type given as the basetype, as an argument.
For cases like the one below, PG creates the aggregate with no args.
```sql
CREATE AGGREGATE oldcnt (
   sfunc = int8inc, basetype = 'ANY', stype = int8,
   initcond = '0'
);
```

In the first example, `defItem->arg` is a `TypeName` and we pass it as an arg to search for the local aggregate created on the coordinator.